### PR TITLE
Action for wait for page

### PR DESCRIPTION
**Pull request by maZahaca**
- Added `type` for simple rule with possible values `array` or `string` to make it possible to return array values.
- Added `type` action to type information on the page.
- Added option `waitForPage` for actions to wait for page load after action perform, which generates change of page.
- Added `conditionalActions` to create action conditions.
- Added `exist` action to check if element exist on the page.
- Added ability to add custom actions and transformations

Last part of #18 
Added #25

### CODE REVIEW

**Comment by maZahaca** [comment]
@jifeon for review

File: lib/Parser.js [hunk 1]
```diff
@@ -147,6 +149,10 @@ Parser.prototype = {
             });
     },
 
+    addAction: function () {
```
Comment (jifeon) [inline lib/Parser.js]: need to add jsdoc, it's public

File: lib/Parser.js [hunk 2]
```diff
@@ -369,10 +376,20 @@ Parser.prototype = {
...
             debug('._parseSimpleRule() result %o', results);
+
+            if (rule.type === 'array') {
+                return results;
+            }
+
             return results.map(function (result) {
                 return result.trim();
             }).join(rule.separator || ' ');
         }).then(function (result) {
```
Comment (jifeon) [inline lib/Parser.js]: this `then` is redundant

File: lib/Parser.js [hunk 3]
```diff
@@ -369,10 +376,20 @@ Parser.prototype = {
...
+            if (rule.type === 'array') {
+                return result.map(function (item) {
+                    return this._transforms.produce(rule.transform, item);
+                }.bind(this));
```
Comment (jifeon) [inline lib/Parser.js]: `map` accept context as second argument

**Comment by jifeon** [comment]
LGTM!

File: lib/Parser.js (hunk 2 above)
Comment (maZahaca) [inline lib/Parser.js]: What do you mean?

**Comment by maZahaca** [comment]
Good point, I'll fix it

**Pull request merged**